textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/keywordmatch/KeywordMatcher.java

```diff
@@ -124,10 +140,62 @@ public ITuple getNextTuple() throws DataFlowException {
                     }
 
                     boolean allTokenPresent = areAllTrue(tokensPresent);
+                    if(predicate.getOperatorType() == KeywordOperatorType.BASIC) {
```

Comment:
This comment is for line 137
```
if (queryTokens.contains(span.getKey()))
```

I don't understand its logic.  We already did `spanForThisField.add(span);`.  Why do we need to check this `contains()` again?  Should this check be done before `spanForThisField.add(span)`?  Maybe an example in the code can help?